src/uniprotkb/config/UniProtKBColumnConfiguration.tsx

```diff
@@ -703,31 +707,77 @@ UniProtKBColumnConfiguration.set(UniProtKBColumn.ccInteraction, {
     const interactionComments = data[EntrySection.Interaction].commentsData.get(
       'INTERACTION'
     ) as InteractionComment[];
+    const { primaryAccession } = data;
+
+    const interactionDataMap = new Map();
+
+    interactionComments?.forEach((interactionCC) =>
+      interactionCC.interactions.forEach((interaction) => {
+        if (
+          interaction.type === InteractionType.SELF ||
+          (interaction.interactantOne.uniProtKBAccession === primaryAccession &&
+            interaction.interactantTwo.uniProtKBAccession === primaryAccession)
+        ) {
+          interactionDataMap.set('self', 'self');
+        } else if (
+          interaction.interactantOne.uniProtKBAccession === primaryAccession &&
+          interaction.interactantTwo.uniProtKBAccession
+        ) {
+          interactionDataMap.set(
+            interaction.interactantTwo.uniProtKBAccession,
+            interaction.interactantTwo
+          );
+        } else if (
+          interaction.interactantTwo.uniProtKBAccession === primaryAccession &&
+          interaction.interactantOne.uniProtKBAccession
+        ) {
+          interactionDataMap.set(
+            interaction.interactantOne.uniProtKBAccession,
+            interaction.interactantOne
+          );
+        }
+        // Some of the interactions are second level so don't feature the primaryAccession
+        // just ignore them.
+      })
+    );
+
+    const sortedInteractions = Array.from(interactionDataMap.values()).sort(
+      (a, b) => {
+        if (typeof a !== 'string' && typeof b !== 'string') {
+          return (
+            (a as Interactant).geneName?.localeCompare(
+              (b as Interactant).geneName || ''
+            ) || -1
+          );
+        }
+        return -2;
+      }
+    );
```

Comment:
It would be good to specify what is the sorting algorithm here, in plain English in a comment. As far as I can see in the implementation it gives:
- All without a gene name (in the order returned by the backend)
- "Itself"
- All with a gene name (in the order return by the backend)
Is that correct and expected?

Reply:
In the end decided to add a test case - works as a better explanation than a comment.

Reply:
<img width="358" alt="Screenshot 2021-07-19 at 18 00 12" src="https://user-images.githubusercontent.com/6025561/126190597-0ae59368-5675-48ec-a058-bb23d095a570.png">
I'm still seeing this (for this search http://localhost:8080/uniprotkb?query=%22Q00266%22) which seems inconsistent that's why I was asking for the ordering in the first place.

Reply:
Possible change here https://github.com/ebi-uniprot/uniprot-website/pull/419

gives me this in the previous example search

<img width="485" alt="Screenshot 2021-07-19 at 18 25 02" src="https://user-images.githubusercontent.com/6025561/126193914-f9a23aa8-baac-48d0-85b3-67f7ac892907.png">